shared/modules/transaction.utils.ts

```diff
@@ -326,3 +327,37 @@ export function hasTransactionData(transactionData?: Hex): boolean {
     transactionData?.length && transactionData?.toLowerCase?.() !== '0x',
   );
 }
+
+export function parseApprovalTransactionData(data: Hex):
+  | {
+      amountOrTokenId?: BigNumber;
+      isApproveAll?: boolean;
+      isRevokeAll?: boolean;
+    }
+  | undefined {
+  const transactionDescription = parseStandardTokenTransactionData(data);
+  const { args, name } = transactionDescription ?? {};
+
+  if (
+    !['approve', 'increaseAllowance', 'setApprovalForAll'].includes(name ?? '')
```

Comment:
Perhaps we can compare token method names by leveraging the `TransactionType`s like we do in `determineTransactionType`, instead of hardcoding the strings?

Reply:
We technically could, but it's coincidence that the types match the names as some of the transaction types for example don't align with any RPC methods.

Though I'm not opposed to using a central RPC method enum, assuming one exists or we add one.